EIPS/eip-7640.md

```diff
@@ -0,0 +1,228 @@
+---
+eip: 7640
+title: Transaction Revert Protection
+description: Defines a new transaction type which rejects reverted transactions
+author: Joseph Poon (@josephpoon), Christopher Jeffrey (@chjj), Boyma Fahnbulleh (@boymanjor)
+discussions-to: https://ethereum-magicians.org/t/eip-7640-transaction-revert-protection/19000
+status: Draft
+type: Standards Track
+category: Core
+created: 2024-02-28
+requires: 140, 141, 658, 1559, 2718, 2930
+---
+
+## Abstract
+
+A consensus rule to flag a reverted transaction as invalid, i.e., ineligible for
+block inclusion, providing consensus layer assurance that users do not pay gas
+fees for unsuccessful transaction execution.
+
+[EIP-658](./eip-658.md) designates a transaction as "failed" if its receipt
+contains a status code of `0x0`. This status code indicates that execution of
+the transaction resulted in error, and the proposed state updates were
+reverted. However, the transaction was still included in a block, and the
+sender paid gas fees for a transaction that had no meaningful effect on the
+state of the network.
+
+We introduce a new [EIP-2718](./eip-2718.md) transaction type, with the format
+`0x05 || rlp([revert_protect, chain_id, nonce, max_priority_fee_per_gas,
+max_fee_per_gas, gas_limit, destination, amount, data, access_list,
+signature_y_parity, signature_r, signature_s])`.
```

Comment:
This might be a bit too technical to add in the Abstract. Might be better to move it to Specification.